src/Roslyn.Diagnostics.Analyzers/UnitTests/ExportedPartsShouldHaveImportingConstructorTests.cs

```diff
@@ -39,6 +39,35 @@ class C {{
             }.RunAsync();
         }
 
+        [Theory]
+        [InlineData("System.Composition")]
+        [InlineData("System.ComponentModel.Composition")]
+        public async Task SingleExpectedConstructor_PrimaryConstructors_CSharpAsync(string mefNamespace)
+        {
+            var source = $$"""
+                using {{mefNamespace}};
+                using System.Diagnostics.CodeAnalysis;
+
+                [Export]
+                [method: SuppressMessage("RoslynDiagnosticsReliability", "RS0034:Exported parts should have [ImportingConstructor]", Justification = "Used incorrectly by tests")]
+                class C(string s)
```

Comment:
@sharwell @mavasani The testign harness doesn't parse this properly (maybe it's using an older version of roslyn?). Could you let me know what i need to update to get primary constructors understood by roslyn-analyzers tests?

Reply:
➡️ This is now resolved